### PR TITLE
For #19976: Adjust telemetry for credit cards.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
I think we definitely want to check with someone on the Data Engineering side before we carry this out. It's not obvious to me if they wanted this done after the fact that they already setup looker to check the existing metrics.

**Comment by gabrielluong** [comment]
cc @wlach for feedback

**Comment by AndiAJ** [comment]
Hi, 

I've just checked @mcarare 's changes on a custom build and this is how the credit card related counters got generated:

```
"counter": {
          "credit_cards.autofill_card": 2,
          "credit_cards.delete_card": 1,
          "credit_cards.manual_save_card": 1,
          "events.normal_and_private_uri_count": 2,
          "events.total_uri_count": 2,
          "glean.validation.foreground_count": 4,
          "logins_store.unlock_count": 2,
          "metrics.tabs_open_count": 1,
          "metrics.top_sites_count": 3,
          "places_manager.read_query_count": 6,
          "places_manager.write_query_count": 3
        }

```

Also made a comparison between the counters introduced in [#18711 (left side)](https://github.com/mozilla-mobile/fenix/issues/18711#issuecomment-852991221) and how are they generated now (rightside)
image

@wlach please review and share your thoughts. ☺️

**Comment by wlach** [comment]
> I think we definitely want to check with someone on the Data Engineering side before we carry this out. It's not obvious to me if they wanted this done after the fact that they already setup looker to check the existing metrics. If we also did land this, I imagine we will also have to uplift it to beta since credit cards are slated to be released for 90. The timing isn't great since most people will be away soon.

👍 from me for landing this change: I think using the `credit_cards` category will make the metrics easier to find in the future. We actually already have the columns in BigQuery from when it originally landed in #18711.

Because of the way our history works, we will always have the `metrics.*` versions of the columns in our data warehouse, but we can easily hide them from Looker and other tools once we implement https://github.com/mozilla/probe-scraper/issues/320

File: app/metrics.yaml [hunk 1]
```diff
@@ -5840,3 +5786,52 @@ start_on_home:
...
+    type: counter
+    description: |
+      A counter of the number of credit cards that have been saved
+      manually by the user.
+    send_in_pings:
+      - metrics
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/18711
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/19548#issuecomment-848811030
```
Comment (gabrielluong) [inline app/metrics.yaml:5800]: I think we want to keep the 2 data_reviews that are listed prior.

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/controller/CreditCardEditorController.kt [hunk 2]
```diff
@@ -66,7 +67,7 @@ class DefaultCreditCardEditorController(
     }
 
     override fun handleDeleteCreditCard(guid: String) {
-        settings.creditCardsDeletedCount += 1
+        metrics.track(Event.CreditCardDelete)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/creditcards/controller/CreditCardEditorController.kt:70]: Can we also add a verify in the tests?

File: app/metrics.yaml (hunk 1 above)
Comment (mcarare) [reply app/metrics.yaml:5800]: This is the original data review. Not sure what other data reviews you are referring to.

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/controller/CreditCardEditorController.kt (hunk 2 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/creditcards/controller/CreditCardEditorController.kt:70]: I will also add tests. Just wanted to confirm the approach first. This was not marked as ready for review.

**Comment by Amejia481** [comment]
Just to let you know @mcarare, there is a new additional step when new adding metrics, adding the glean annotations for it you can see [the steps in this wiki](https://github.com/mozilla-mobile/fenix/wiki/Add-a-Glean-Annotation-for-an-event).

**Comment by mcarare** [comment]
@Amejia481 Thanks for the heads up, I have seen that wiki entry and I already put up a PR for recent tabs telemetry. I am going to put one for credit cards as soon as this PR gest is approved.

**Comment by Amejia481** [comment]
Awesome, if you need any help just let know :)

**Comment by mcarare** [comment]
Closing as per https://github.com/mozilla-mobile/fenix/pull/20691#issuecomment-893569253.

**Pull request closed without merging**